cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/partitioned/PartitionedFileSetDataset.java

```diff
@@ -614,7 +614,7 @@ public String getInputFormatClassName() {
   protected Collection<String> computeFilterInputPaths() {
     PartitionFilter filter;
     try {
-      filter = PartitionedFileSetArguments.getInputPartitionFilter(runtimeArguments, partitioning);
+      filter = PartitionedFileSetArguments.getInputPartitionFilter(runtimeArguments);
```

Comment:
I suppose that can happen if user improperly constructs a PartitionFilter (with extra fields).
If we want to silently handle this (which was happening before), we can update PartitionFilter#update method to iterate over the partition key's fields instead of iterating over the conditions:
https://github.com/caskdata/cdap/blob/fc86cfbb822ccc5501138310d9f360ca57305b0b/cdap-api/src/main/java/co/cask/cdap/api/dataset/lib/PartitionFilter.java#L60-L68

Though personally, I don't think we should silently handle that.